src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,473 @@
+
+            if self.complex_type is not "nerve" or self.cover is not "functional": # Nerve + functional cover is a special case where it's better to use Mapper than C++ code
+
+                ct = "GIC" if self.complex_type == "gic" else "Nerve"
+                self.complex = NGIComplex()
+                self.complex.set_type(ct)
+                self.complex.set_distance_matrix_name(self.distance_matrix_name)
+                self.complex.set_verbose(self.verbose)
+
+                if self.input_type == "point cloud":
+                    self.complex.set_point_cloud_from_range(X)
+                elif self.input_type == "distance matrix":
+                    self.complex.set_distances_from_range(X)
+            
+                self.complex.set_color_from_range(self.colors)
+            
+                if self.complex_type == "gic":
+                    if self.graph == "rips":
+                        if self.rips_threshold is not None:
+                            self.complex.set_graph_from_rips(self.rips_threshold)
+                        else:
+                            self.complex.set_subsampling(self.C, self.beta)
+                            self.complex.set_graph_from_automatic_rips(self.N)
+
+            if self.cover == "voronoi":
+                assert self.complex_type is not "nerve"
+                self.complex.set_cover_from_Voronoi(self.voronoi_samples)
+
+            elif self.cover == "functional":
+
+                if self.complex_type == "gic":
+
+                    self.complex.set_function_from_range(self.filters)
+
+                    if self.resolutions is None:
+                        self.complex.set_automatic_resolution()
+                    else:
+                        self.complex.set_resolution_with_interval_number(self.resolutions[0])
+
+                    if self.gains is None:
+                        self.complex.set_gain(.33)
+                    else:
+                        self.complex.set_gain(self.gains[0])
+
+                    self.complex.set_cover_from_function()
+
+                elif self.complex_type == "nerve": # it's actually better to use Mapper with constant clustering
+                    self.complex = _MapperComplex(filters=self.filters, filter_bnds=self.filter_bnds, colors=self.colors, 
+                                                 resolutions=self.resolutions, gains=self.gains, inp=self.input_type, 
+                                                 clustering=self._constant_clustering, mask=self.mask, N=self.N, beta=self.beta, C=self.C)
+                    self.complex.fit(X)
+                    self.simplex_tree = self.complex.mapper_
+                    self.node_info = self.complex.node_info_
+
+            elif self.cover == "precomputed":
+                self.complex.set_cover_from_range(self.assignments)
+
+            if self.complex_type is not "nerve" or self.cover is not "functional":
```

Comment:
`is not` should only be used for `None`, for instance. Use `!=` operator for literals.
```suggestion
            if self.complex_type != "nerve" or self.cover != "functional":
```

Reply:
Thanks! Done.